### PR TITLE
Review types for grid components

### PR DESCRIPTION
**Pull request by IlyaKhD**

### CODE REVIEW

File: js/ui/data_grid.d.ts [hunk 1]
```diff
@@ -69,7 +74,406 @@ import Widget, {
...
+     * @type_function_param1_field4 itemElement:dxElement
+     * @type_function_param1_field5 fromIndex:number
+     * @type_function_param1_field6 toIndex:number
+     * @type_function_param1_field7 fromComponent:dxSortable|dxDraggable
+     * @type_function_param1_field8 toComponent:dxSortable|dxDraggable
+     * @type_function_param1_field9 fromData:any
+     * @type_function_param1_field10 toData:any
+     * @type_function_param1_field11 dropInsideItem:boolean
+     */
+    onAdd?: ((e: ComponentRowDraggingEvent<T> & DragDropInfo) => void),
```
Comment (vconst) [inline js/ui/data_grid.d.ts:354]: Why is not the AddEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 2]
```diff
@@ -69,7 +74,406 @@ import Widget, {
...
+     * @type_function_param1_field5 itemElement:dxElement
+     * @type_function_param1_field6 fromIndex:number
+     * @type_function_param1_field7 toIndex:number
+     * @type_function_param1_field8 fromComponent:dxSortable|dxDraggable
+     * @type_function_param1_field9 toComponent:dxSortable|dxDraggable
+     * @type_function_param1_field10 fromData:any
+     * @type_function_param1_field11 toData:any
+     * @type_function_param1_field12 dropInsideItem:boolean
+     */
+    onDragChange?: ((e: Cancelable & ComponentRowDraggingEvent<T> & DragDropInfo) => void),
```
Comment (vconst) [inline js/ui/data_grid.d.ts:372]: Why is not the DragEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 3]
```diff
@@ -69,7 +74,406 @@ import Widget, {
...
+     * @type_function_param1_field5 itemElement:dxElement
+     * @type_function_param1_field6 fromIndex:number
+     * @type_function_param1_field7 toIndex:number
+     * @type_function_param1_field8 fromComponent:dxSortable|dxDraggable
+     * @type_function_param1_field9 toComponent:dxSortable|dxDraggable
+     * @type_function_param1_field10 fromData:any
+     * @type_function_param1_field11 toData:any
+     * @type_function_param1_field12 dropInsideItem:boolean
+     */
+    onDragEnd?: ((e: Cancelable & ComponentRowDraggingEvent<T> & DragDropInfo) => void),
```
Comment (vconst) [inline js/ui/data_grid.d.ts:390]: Why is not the DragEndEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 4]
```diff
@@ -69,7 +74,406 @@ import Widget, {
...
+     * @type_function_param1_field5 itemElement:dxElement
+     * @type_function_param1_field6 fromIndex:number
+     * @type_function_param1_field7 toIndex:number
+     * @type_function_param1_field8 fromComponent:dxSortable|dxDraggable
+     * @type_function_param1_field9 toComponent:dxSortable|dxDraggable
+     * @type_function_param1_field10 fromData:any
+     * @type_function_param1_field11 toData:any
+     * @type_function_param1_field12 dropInsideItem:boolean
+     */
+    onDragMove?: ((e: Cancelable & ComponentRowDraggingEvent<T> & DragDropInfo) => void),
```
Comment (vconst) [inline js/ui/data_grid.d.ts:408]: Why is not the DragMoveEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 5]
```diff
@@ -69,7 +74,406 @@ import Widget, {
...
+     * @type_function_param1_field3 itemData:any
+     * @type_function_param1_field4 itemElement:dxElement
+     * @type_function_param1_field5 fromIndex:number
+     * @type_function_param1_field6 toIndex:number
+     * @type_function_param1_field7 fromComponent:dxSortable|dxDraggable
+     * @type_function_param1_field8 toComponent:dxSortable|dxDraggable
+     * @type_function_param1_field9 fromData:any
+     * @type_function_param1_field10 toData:any
+     */
+    onRemove?: ((e: ComponentRowDraggingEvent<T>) => void),
```
Comment (vconst) [inline js/ui/data_grid.d.ts:437]: Why is not the RemoveEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 6]
```diff
@@ -69,7 +74,406 @@ import Widget, {
...
+     * @type_function_param1_field5 fromIndex:number
+     * @type_function_param1_field6 toIndex:number
+     * @type_function_param1_field7 fromComponent:dxSortable|dxDraggable
+     * @type_function_param1_field8 toComponent:dxSortable|dxDraggable
+     * @type_function_param1_field9 fromData:any
+     * @type_function_param1_field10 toData:any
+     * @type_function_param1_field11 dropInsideItem:boolean
+     * @type_function_param1_field12 promise:Promise<void>
+     */
+    onReorder?: ((e: ComponentRowDraggingEvent<dxDataGrid> & DragReorderInfo) => void),
```
Comment (vconst) [inline js/ui/data_grid.d.ts:455]: Why is not the ReorderEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 7]
```diff
@@ -114,103 +518,18 @@ export interface GridBaseOptions<T = GridBase> extends WidgetOptions<T> {
...
-        rightPosition?: string,
-        /**
-         * @docid
-         * @prevFileNamespace DevExpress.ui
-         * @default "Unfix"
-         */
-        unfix?: string
-      }
-    };
+    columnFixing?: GridBaseColumnFixing;
```
Comment (vconst) [inline js/ui/data_grid.d.ts]: Rename GridBaseColumnFixing to ColumnFixing

File: js/ui/data_grid.d.ts [hunk 8]
```diff
@@ -523,168 +674,32 @@ export interface GridBaseOptions<T = GridBase> extends WidgetOptions<T> {
...
-      enterKeyAction?: 'startEdit' | 'moveFocus',
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @type Enums.GridEnterKeyDirection
-       * @default "none"
-       */
-      enterKeyDirection?: 'none' | 'column' | 'row'
-    };
+    keyboardNavigation?: GridBaseKeyboardNavigation;
```
Comment (vconst) [inline js/ui/data_grid.d.ts]: Rename GridBaseKeyboardNavigation to KeyboardNavigation

File: js/ui/data_grid.d.ts [hunk 9]
```diff
@@ -696,108 +711,141 @@ export interface GridBaseOptions<T = GridBase> extends WidgetOptions<T> {
...
+     * @type_function_param1_field1 component:this
+     * @type_function_param1_field2 element:TElement
+     * @type_function_param1_field3 model:any
      * @type_function_param1_field4 formOptions:object
      * @action
      * @prevFileNamespace DevExpress.ui
      * @public
      */
-    onAdaptiveDetailRowPreparing?: ((e: { component?: T, element?: TElement, model?: any, formOptions?: any }) => void);
+    onAdaptiveDetailRowPreparing?: ((e: ComponentEvent<T> & AdaptiveDetailRowPreparingInfo) => void);
```
Comment (vconst) [inline js/ui/data_grid.d.ts:722]: Why is not AdaptiveDetailRowPreparingEvent<T> here?

File: js/ui/data_grid.d.ts [hunk 10]
```diff
@@ -696,108 +711,141 @@ export interface GridBaseOptions<T = GridBase> extends WidgetOptions<T> {
...
+     * @type_function_param1_field1 component:this
+     * @type_function_param1_field2 element:TElement
+     * @type_function_param1_field3 model:any
      * @type_function_param1_field4 error:Error
      * @action
      * @prevFileNamespace DevExpress.ui
      * @public
      */
-    onDataErrorOccurred?: ((e: { component?: T, element?: TElement, model?: any, error?: Error }) => void);
+    onDataErrorOccurred?: ((e: ComponentEvent<T> & DataErrorOccurredInfo) => void);
```
Comment (vconst) [inline js/ui/data_grid.d.ts:735]: Why is not DataErrorOccurred<T> here?

File: js/ui/data_grid.d.ts [hunk 11]
```diff
@@ -3734,15 +4525,18 @@ export interface dxDataGridEditing extends GridBaseEditing {
...
     /**
      * @docid dxDataGridOptions.editing.texts
      * @prevFileNamespace DevExpress.ui
      * @public
      */
     texts?: any;
 }
-export interface dxDataGridScrolling extends GridBaseScrolling {
+
+export type dxDataGridScrolling = Scrolling;
```
Comment (vconst) [inline js/ui/data_grid.d.ts:4537]: Where is deprecated doc comment?

File: js/ui/data_grid.d.ts [hunk 12]
```diff
@@ -3752,7 +4546,10 @@ export interface dxDataGridScrolling extends GridBaseScrolling {
      */
     mode?: 'infinite' | 'standard' | 'virtual';
 }
-export interface dxDataGridSelection extends GridBaseSelection {
+
+export type dxDataGridSelection = Selection;
```
Comment (vconst) [inline js/ui/data_grid.d.ts:4550]: Where is deprecated doc comment?

File: js/ui/data_grid.d.ts [hunk 13]
```diff
@@ -3999,50 +4796,52 @@ declare class dxDataGrid extends Widget implements GridBase {
     updateDimensions(): void;
 }
 
+export type dxDataGridColumn = Column;
```
Comment (vconst) [inline js/ui/data_grid.d.ts:4799]: Where is deprecated doc comment?

File: js/ui/data_grid.d.ts [hunk 14]
```diff
@@ -4198,80 +4999,81 @@ export interface dxDataGridColumnButton extends GridBaseColumnButton {
      * @prevFileNamespace DevExpress.ui
      * @public
      */
-    visible?: boolean | ((options: { component?: dxDataGrid, row?: dxDataGridRowObject, column?: dxDataGridColumn }) => boolean);
+    visible?: boolean | ((options: { component?: dxDataGrid, row?: RowObject, column?: Column }) => boolean);
 }
 
+export type dxDataGridRowObject = RowObject;
```
Comment (vconst) [inline js/ui/data_grid.d.ts:5005]: Where is deprecated doc comment?

File: js/ui/data_grid.d.ts (hunk 8 above)
Comment (IlyaKhD) [reply js/ui/data_grid.d.ts]: done

File: js/ui/data_grid.d.ts (hunk 7 above)
Comment (IlyaKhD) [reply js/ui/data_grid.d.ts]: done

**Review by NickMitrokhin — APPROVED** [review]

**Pull request merged**